pkg/controllers/work/apply_controller.go

```diff
@@ -536,6 +544,62 @@ func trackJobAvailability(curObj *unstructured.Unstructured) (ApplyAction, error
 	return manifestNotTrackableAction, nil
 }
 
+func trackServiceAvailability(curObj *unstructured.Unstructured) (ApplyAction, error) {
+	var service v1.Service
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &service); err != nil {
+		return errorApplyAction, controller.NewUnexpectedBehaviorError(err)
+	}
+	switch service.Spec.Type {
+	case "":
+		fallthrough //default service type is ClusterIP
+	case v1.ServiceTypeClusterIP:
+		fallthrough
+	case v1.ServiceTypeNodePort:
```

Comment:
user can set their own Nodeport so the exsitence of the nodePort doesn't really mean it's ready.

https://kubernetes.io/docs/concepts/services-networking/service/#nodeport-custom-port

Reply:
it's similar to clusterIP of the clusterIP type and it can be customized too. The existence of the clusterIP does not mean it's accessible.

Should we handle them in a consistent way?

Reply:
The reason the ClusterIPs field is a good signal is that the values there are guaranteed to be correct (users can't set bad IPs).  Let me check if the nodePort field also has this property.